tests/har.spec.ts

```diff
@@ -281,3 +281,33 @@ it('should have popup requests', async ({ contextFactory, server }, testInfo) =>
   expect(entries[1].request.url).toBe(server.PREFIX + '/one-style.css');
   expect(entries[1].response.status).toBe(200);
 });
+
+it('should have connection details', async ({ contextFactory, server, browserName, platform }, testInfo) => {
+  const { page, getLog } = await pageWithHar(contextFactory, testInfo);
+  await page.goto(server.EMPTY_PAGE);
+  const log = await getLog();
+  const { serverIPAddress, _serverPort: port, _securityDetails: securityDetails } = log.entries[0];
+  expect(serverIPAddress).toMatch(/^127\.0\.0\.1|\[::1\]/);
+  expect(port).toBe(server.PORT);
+  if (browserName === 'webkit' && platform === 'darwin')
+    expect(securityDetails).toEqual({protocol: 'SSL 2.0'});
```

Comment:
WebKit on mac thinks that 127.0.0.1 is SSL? This doesn't match what I see in Safari.

![image](https://user-images.githubusercontent.com/4624233/120503602-4c634c80-c378-11eb-87af-77acca23e3da.png)

Reply:
Oops…this should probably be marked failing for now. It looks like we are getting placeholder values (`SSL 2.0` and `TLS_NULL_WITH_NULL_NULL `) over the wire from WebKit protocol:

(scroll over all the way over to `securityConnection` section)

```json
{
  "method": "Target.dispatchMessageFromTarget",
  "params": {
    "targetId": "page-8",
    "message": "{\"method\":\"Network.loadingFinished\",\"params\":{\"requestId\":\"9.1\",\"timestamp\":0.027748466171033215,\"metrics\":{\"protocol\":\"http/1.1\",\"priority\":\"high\",\"remoteAddress\":\"127.0.0.1:8000\",\"connectionIdentifier\":\"093BB7CD-C601-4A3A-B68F-0654B44123B1\",\"requestHeaders\":{\"Accept\":\"text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8\",\"Upgrade-Insecure-Requests\":\"1\",\"Host\":\"127.0.0.1:8000\",\"User-Agent\":\"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/14.2 Safari/605.1.15\",\"Accept-Language\":\"en-us\",\"Accept-Encoding\":\"gzip, deflate\",\"Connection\":\"keep-alive\"},\"requestHeaderBytesSent\":354,\"requestBodyBytesSent\":0,\"responseHeaderBytesReceived\":184,\"responseBodyBytesReceived\":99,\"responseBodyDecodedSize\":99,\"securityConnection\":{\"protocol\":\"SSL 2.0\",\"cipher\":\"TLS_NULL_WITH_NULL_NULL\"}}}}"
  },
  "browserContextId": "8000000000000002",
  "pageProxyId": "7"
}
```

Reply:
This code from WebInspector might be relevant:

```js
    function loadedSecurely()
    {
        if (this.urlComponents.scheme !== "https" && this.urlComponents.scheme !== "wss" && this.urlComponents.scheme !== "sftp")
            return false;
        if (isNaN(this._timingData.secureConnectionStart) && !isNaN(this._timingData.connectionStart))
            return false;
        return true;
    }
```